src/libopensc/iasecc-sdo.c

```diff
@@ -204,19 +204,26 @@ iasecc_sdo_free(struct sc_card *card, struct iasecc_sdo *sdo)
 
 
 static int
-iasecc_crt_parse(struct sc_card *card, unsigned char *data, struct iasecc_se_info *se)
+iasecc_crt_parse(struct sc_card *card, unsigned char *data, size_t data_len, struct iasecc_se_info *se)
 {
 	struct sc_context *ctx = card->ctx;
 	struct sc_crt crt;
 	int ii, offs, len, parsed_len = -1;
 
 	sc_log(ctx, "iasecc_crt_parse(0x%X) called", *data);
 
+	if (data_len < 2)
+		LOG_FUNC_RETURN(ctx, SC_ERROR_INVALID_DATA);
+
 	memset(&crt, 0, sizeof(crt));
 	crt.tag = *(data + 0);
 	len = *(data + 1);
+	if ((size_t) len != data_len - 2)
+		LOG_FUNC_RETURN(ctx, SC_ERROR_INVALID_DATA);
```

Comment:
does this work the way that the fixed block is parsed always or there can be several follow-up blocks to parse? I would say the second as it is called in cycle so I think this check is not correct.

Reply:
Makes sense, thanks for review. Will fix